packages/account/src/Components/poi/idv-status/idv-failed/idv-failed.tsx

```diff
@@ -0,0 +1,395 @@
+    TIDVErrorStatus,
+    IDV_NOT_APPLICABLE_OPTION,
+    isMobile,
+} from '@deriv/shared';
+import { Localize, localize } from '@deriv/translations';
+import PoiNameExample from 'Assets/ic-poi-name-example.svg';
+import PoiDobExample from 'Assets/ic-poi-dob-example.svg';
+import PoiNameDobExample from 'Assets/ic-poi-name-dob-example.svg';
+import FormBody from 'Components/form-body';
+import PersonalDetailsForm from 'Components/forms/personal-details-form';
+import LoadErrorMessage from 'Components/load-error-message';
+import {
+    getIDVDocumentType,
+    isAdditionalDocumentValid,
+    isDocumentNumberValid,
+    isDocumentTypeValid,
+    makeSettingsRequest,
+    shouldHideHelperImage,
+    validate,
+    validateName,
+} from 'Helpers/utils';
+import { TIDVForm, TPersonalDetailsForm } from 'Types';
+import FormSubHeader from 'Components/form-sub-header';
+import IDVForm from 'Components/forms/idv-form';
+import classNames from 'classnames';
+import FormFooter from 'Components/form-footer';
+
+type TRestState = {
+    api_error: string;
+    errors?: boolean;
+    form_initial_values: TIdvFailedForm;
+    changeable_fields: string[];
+};
+
+type TIdvFailed = {
+    account_settings: GetSettings;
+    getChangeableFields: () => string[];
+    handleSubmit: () => void;
+    is_from_external: boolean;
+    mismatch_status: TIDVErrorStatus;
+    residence_list: ResidenceList;
+    latest_status: DeepRequired<GetAccountStatus>['authentication']['attempts']['latest'];
+};
+
+type TIDVFailureConfig = {
+    required_fields: string[];
+    side_note_image: JSX.Element;
+    failure_message: React.ReactNode;
+    inline_note_text: React.ReactNode;
+};
+
+type TIdvFailedForm = Partial<TIDVForm> & Partial<TPersonalDetailsForm>;
+
+const IdvFailed = ({
+    getChangeableFields,
+    is_from_external,
+    residence_list,
+    account_settings,
+    handleSubmit,
+    mismatch_status = idv_error_statuses.poi_failed,
+    latest_status,
+}: TIdvFailed) => {
+    const [idv_failure, setIdvFailure] = React.useState<TIDVFailureConfig>({
+        required_fields: [],
+        side_note_image: <PoiNameDobExample />,
+        failure_message: null,
+        inline_note_text: null,
+    });
+    const [is_loading, setIsLoading] = React.useState(true);
+    const [rest_state, setRestState] = React.useState<TRestState>({
+        api_error: '',
+        errors: false,
+        form_initial_values: {},
+        changeable_fields: [],
+    });
+
+    const is_document_upload_required = React.useMemo(
+        () => [idv_error_statuses.poi_expired, idv_error_statuses.poi_failed].includes(mismatch_status),
+        [mismatch_status]
+    );
+
+    const citizen = account_settings?.citizen;
+    const selected_country = residence_list.find(residence_data => residence_data.value === citizen) || {};
+
+    React.useEffect(() => {
+        const document_name = getIDVDocumentType(latest_status, selected_country);
+        const generateIDVError = () => {
```

Comment:
Isn't it better to extract this function from useEffect as it's going to be generated on each render or whenever any of the dependency changes 

we can extract and memoize it

Reply:
I could just wrap it with callback. Because after extraction it should be added to dependency

Reply:
Yes it needs to defined in the dependency list however it'll be a stable one